### ISSUE TITLE
LLD fails to link objects from binutils upcoming 2.41 on RISCV

### ISSUE THREAD

**Issue by kraj** [llvm/llvm-project#64102]
I have seeing the below error with lld ( trunk and 16.0.0 release )on riscv in combination with GAS from binutils 2.41 ( upcoming ) release built objects e.g. from glibc .a files.

```
riscv64-yoe-linux-ld.lld: error: TOPDIR/build/tmp/work/riscv64-yoe-linux/libcap-ng/0.8.3-r0/recipe-sysroot/usr/lib/libc_nonshared.a(pthread_atfork.oS):(.debug_loclists+0xd): unknown relocation (60) against symbol .LVL0

riscv64-yoe-linux-ld.lld: error: TOPDIR/build/tmp/work/riscv64-yoe-linux/libcap-ng/0.8.3-r0/recipe-sysroot/usr/lib/libc_nonshared.a(pthread_atfork.oS):(.debug_loclists+0xd): unknown relocation (61) against symbol .Ltext0
```

These are `R_RISCV_SET_ULEB128` and `R_RISCV_SUB_ULEB128` relocations.

**Comment by jrtc27** [comment]
https://reviews.llvm.org/D142880

**Comment by asb** [comment]
Thanks for reporting - we were aware of needing this to land but really should have made an issue to track that.

**Comment by kraj** [comment]
> https://reviews.llvm.org/D142880

and this one too - https://reviews.llvm.org/D142879

**Comment by MaskRay** [comment]
The first step is my assembler change: https://reviews.llvm.org/D157657 
Then we need to change lld.

**Comment by asb** [comment]
@MaskRay this can be closed now, right?

**Comment by MaskRay** [comment]
Yes! Fixed after #72610 and #77261
